core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala

```diff
@@ -642,7 +642,7 @@ class DynamicBrokerReconfigurationTest extends QuorumTestHarness with SaslSetup
     props.put(KafkaConfig.CompressionTypeProp, "gzip")
     props.put(KafkaConfig.LogPreAllocateProp, true.toString)
     props.put(KafkaConfig.LogMessageTimestampTypeProp, TimestampType.LOG_APPEND_TIME.toString)
-    props.put(KafkaConfig.LogMessageTimestampDifferenceMaxMsProp, "1000")
```

Comment:
I have made three types of changes to teh tests
1) Added parametrized tests to validation on all new configs and the deprecated one. See changes in `PlaintextProducerSendTest.scala` and `ProducerRequestTest.scala`
2) Added test cases for backward compatibility for both `before` and `after` cases
3) Removed unnecessary calls to set `TopicConfig.MESSAGE_TIMESTAMP_DIFFERENCE_MAX_MS_CONFIG` to the default value. This calls don't add any value to the test since the calls are not changing the default value. See example changes in `LogCleanerTest.scala` and `LogManagerTest.scala`